File: src/utils/currency.ts
// Currency formatting helper. The symbol is supplied by PreferencesContext so
// the whole app can switch currency display from one place.

export interface FormatCurrencyOptions {
    symbol?: string;
    maximumFractionDigits?: number;
    showSymbol?: boolean;
}

export const formatCurrency = (
    value: number | string | null | undefined,
    {symbol = '₹', maximumFractionDigits = 2, showSymbol = true}: FormatCurrencyOptions = {},
): string => {
    const num = Number(value);
    const safe = Number.isFinite(num) ? num : 0;
    const sign = safe < 0 ? '-' : '';
    const abs = Math.abs(safe);

    // Indian grouping (1,00,000) matches the app's INR default; still correct
    // for other symbols since it only affects digit grouping.
    const grouped = abs.toLocaleString('en-IN', {
        maximumFractionDigits,
        minimumFractionDigits: 0,
    });

    return `${sign}${showSymbol ? symbol : ''}${grouped}`;
};

File: lib/core/utils/currency.dart
import 'package:intl/intl.dart';

/// Currency formatter ported from `legacy_rn/src/utils/currency.ts`.
/// Uses Indian digit grouping (1,00,000) to match the INR default.
String formatCurrency(
  num? value, {
  String symbol = '₹',
  int maximumFractionDigits = 2,
  bool showSymbol = true,
}) {
  final v = (value ?? 0).toDouble();
  final sign = v < 0 ? '-' : '';
  final formatter = NumberFormat.decimalPattern('en_IN')
    ..maximumFractionDigits = maximumFractionDigits
    ..minimumFractionDigits = 0;
  return '$sign${showSymbol ? symbol : ''}${formatter.format(v.abs())}';
}
